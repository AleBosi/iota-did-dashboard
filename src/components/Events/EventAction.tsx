import React from "react";
import { Event } from "../../models/event";

interface Props {
  event: Event;
  onExecute: (eventId: string) => void;
}

export default function EventAction({ event, onExecute }: Props) {
  return (
    <div className="flex items-center gap-2 my-2">
      <span>
        <b>{event.type}</b>: {event.description}
      </span>
      {!event.done && (
        <button
          className="bg-blue-500 text-white px-3 py-1 rounded"
          onClick={() => onExecute(event.id)}
        >
          Esegui evento
        </button>
      )}
      {event.done && <span className="text-green-600 font-bold">Completato</span>}
    </div>
  );
}
