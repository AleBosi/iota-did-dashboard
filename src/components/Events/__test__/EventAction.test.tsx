import { render, screen, fireEvent } from "@testing-library/react";
import EventAction from "../EventAction";

// Mock completo del tipo Event
const mockEvent = {
  id: "evt1",
  type: "Produzione",
  description: "Avvia la macchina per il lotto X",
  productId: "prod-001",
  operatoreId: "op-123",
  macchinarioId: "mac-456",
  creatorId: "creator-999",
  date: "2025-08-05T12:00:00Z",
  done: false
};

describe("EventAction component", () => {
  it("visualizza tipo e descrizione evento", () => {
    render(<EventAction event={mockEvent} onExecute={() => {}} />);
    expect(screen.getByText("Produzione")).toBeInTheDocument();
    expect(screen.getByText(/Avvia la macchina/)).toBeInTheDocument();
  });

  it("mostra il bottone se l'evento NON Ã¨ fatto", () => {
    render(<EventAction event={mockEvent} onExecute={() => {}} />);
    expect(screen.getByText("Esegui evento")).toBeInTheDocument();
    expect(screen.queryByText("Completato")).toBeNull();
  });

  it("NON mostra il bottone e mostra 'Completato' se event.done = true", () => {
    render(<EventAction event={{ ...mockEvent, done: true }} onExecute={() => {}} />);
    expect(screen.queryByText("Esegui evento")).toBeNull();
    expect(screen.getByText("Completato")).toBeInTheDocument();
  });

  it("chiama onExecute con l'id dell'evento quando clicchi il bottone", () => {
    const onExecute = jest.fn();
    render(<EventAction event={mockEvent} onExecute={onExecute} />);
    fireEvent.click(screen.getByText("Esegui evento"));
    expect(onExecute).toHaveBeenCalledWith("evt1");
  });

  it("snapshot del componente", () => {
    const { asFragment } = render(<EventAction event={mockEvent} onExecute={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
