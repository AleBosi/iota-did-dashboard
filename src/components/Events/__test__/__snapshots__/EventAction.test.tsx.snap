// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`EventAction component snapshot del componente 1`] = `
<DocumentFragment>
  <div
    class="flex items-center gap-2 my-2"
  >
    <span>
      <b>
        Produzione
      </b>
      : Avvia la macchina per il lotto X
    </span>
    <button
      class="bg-blue-500 text-white px-3 py-1 rounded"
    >
      Esegui evento
    </button>
  </div>
</DocumentFragment>
`;
